pkg/storage/util/util.go

```diff
@@ -59,6 +62,49 @@ func UpdateCondition(cr *imageregistryv1.Config, conditionType string, status op
 	cr.Status.Conditions = conditions
 }
 
+// GetInfrastructure gets information about the cloud platform that the cluster is
+// installed on including the Type, Region, and other platform specific information.
+// Currently the install config is used as a backup to be compatible with upgrades
+// from 4.1 -> 4.2 when platformStatus did not exist, but should be able to be removed
+// in the future.
```

Comment:
@adambkaplan @coreydaley @dmage so we decided we still need this because a cluster from 4.1 doesn't have the new config info?

We should be pushing on the install team to ensure the new apis are populated during upgrade.  We cannot continue to carry 4.1 specific logic "just in case", we need to be able to assume that a 4.3 cluster looks like any other 4.3 cluster, regardless of what it started life as.

So if there are differences between a 4.1->4.2->4.3 cluster and a 4.3 cluster, those should be bugs against whoever owns the api that is manifesting differences.

Reply:
@bparees yes, we have to carry this until they fix it. There is ticket to fix it: https://issues.redhat.com/browse/CORS-1173

Reply:
perfect, thanks @dmage

Reply:
@bparees @dmage We still need to merge this though.

Reply:
yes, understood. just wanted to make sure we were doing the right thing long term.